joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/AstForGenDeclarationCreator.scala

```diff
@@ -13,12 +15,52 @@ trait AstForGenDeclarationCreator(implicit withSchemaValidation: ValidationMode)
     genDecl.json(ParserKeys.Tok).str match {
       case "import" => astForImport(genDecl)
       case "const"  => List[Ast]()
-      case "type"   => List[Ast]()
+      case "type"   => astForTypeSpec(genDecl)
       case "var"    => astForValueSpec(genDecl)
 
     }
   }
 
+  def astForTypeSpec(genDecl: ParserNodeInfo): Seq[Ast] = {
+    genDecl
+      .json(ParserKeys.Specs)
+      .arr
+      .map(createParserNodeInfo)
+      .flatMap { typeSpec =>
```

Comment:
We are iterating through this list inside all the child methods `astForImprt`, `astForValueSpec`, and now inside `astForTypeSpec`. We should move this iteration loop inside `astForGenDecl` method.